### PR TITLE
DoS & Detection Revamp

### PR DESCRIPTION
**Pull request by Swangeon**
Go rewrite of the DoS attack in Honeyscanner.

### CODE REVIEW

File: README.md
Comment (aristofanischionis) [inline README.md:102]: Let's avoid introducing running honeyscanner with `root` privileges, we are just scanning for open ports, so we can change the nmap flags to only '-p-', this way we can keep running honeyscanner as a normal user.

File: .gitignore
Comment (aristofanischionis) [inline .gitignore:34]: too many attacks attacks, nested folders, can we improve this?

File: README.md
Comment (aristofanischionis) [inline README.md:105]: same about sudo

File: honeyscanner/active_attacks/attack_orchestrator.py
Comment (aristofanischionis) [inline honeyscanner/active_attacks/attack_orchestrator.py:20]: Is this comment still valid?

File: honeyscanner/active_attacks/honeypot_port_scanner/honeypot_port_scanner.py
Comment (aristofanischionis) [inline honeyscanner/active_attacks/honeypot_port_scanner/honeypot_port_scanner.py:64]: this needs to be adjusted as now we will run a port scan only, not scanning with scripts (-sC) and versions (-sV) which are parts of the (-A) flag. Adjustments need to be made.

File: honeyscanner/main.py
Comment (aristofanischionis) [inline honeyscanner/main.py:89]: This is not accurate, maybe go is installed in the system but is not in path, it's more accurate to change the error message to something like go is not found in the path. Also we need to consider go compilation on runtime, I am not sure if it's the best idea. Also, providing the compiled binary is also not the best idea, as it won't work in all architectures. Maybe we can provide for amd64 and arm64 and query the system on runtime and determine which one to use.

**Review by aristofanischionis — CHANGES_REQUESTED** [review]
Hi, thanks for the PR, please review my comments and provide the necessary fixes before merging.

**Comment by aristofanischionis** [comment]
@Swangeon Can you please update README with some FAQs to help users. 
In case of troubles, suggest that they install the latest version of anaconda from here
https://www.anaconda.com/download/success
and that they make sure they use python 3.10
and also if they still face problems with pipenv install, they should try emptying their virtualenvs folder and trying again.

Also, don't we need one more line in Pipfile?
`typing = "*"`
to import the typing library?

**Comment by aristofanischionis** [comment]
DoS attack doesn't seem to be able to take down the honeypot. I am running the Cowrie honeypot in the exact same way I was running it before, with the Python code used before, the honeypot was becoming unavailable, and now with the Go module, not anymore, it is still up.

**Comment by aristofanischionis** [comment]
I received these 2 warnings when trying to run Honeyscanner, did you also experience them? Let's fix them before merging.

Screenshot 2024-07-29 at 14 17 59

**Comment by aristofanischionis** [comment]
The Honeypot port scanner is not working properly for me. 
We can try integrating this PR (https://github.com/honeynet/honeyscanner/pull/38) and removing nmap altogether as it's slow, noisy, and too big for what we need.

Screenshot 2024-07-29 at 14 26 38

**Comment by Swangeon** [comment]
> I received these 2 warnings when trying to run Honeyscanner, did you also experience them? Let's fix them before merging.
> Screenshot 2024-07-29 at 14 17 59

This is going to be an annoying error for now that doesn't effect the project that you can look at more [here](https://github.com/paramiko/paramiko/issues/2419). There are ways to repress the warning, but I don't know if we would want to implement a temporary solution for a temporary problem.

> The Honeypot port scanner is not working properly for me. We can try integrating this PR (#38) and removing nmap altogether as it's slow, noisy, and too big for what we need.
> Screenshot 2024-07-29 at 14 26 38

I haven't gotten this error before so I am not quite sure whats causing it but I will look into it. I'm not quite certain merging this PR with @ZupaPomidorowa's PR would be best for now because this PR would now have to wait for their PR to be merged and functional to work more on this. If you want, I can just move on to the fuzzing upgrades we talked about and come back to this at a later date.

File: README.md
Comment (aristofanischionis) [inline README.md:89]: there is a typo here, correct is to have target-ip not target_ip

File: README.md
Comment (aristofanischionis) [inline README.md:102]: same here

File: README.md
Comment (aristofanischionis) [inline README.md:106]: same here

File: README.md
Comment (aristofanischionis) [inline README.md:125]: i like that!

File: .gitignore
Comment (aristofanischionis) [inline .gitignore:22]: what is this for?

File: .gitignore
Comment (aristofanischionis) [inline .gitignore:33]: what is this for?

**Review by aristofanischionis — CHANGES_REQUESTED** [review]
I am facing some issues while running this version of Honeyscanner
Screenshot 2024-08-22 at 14 53 35
Screenshot 2024-08-22 at 14 53 46
Screenshot 2024-08-22 at 14 53 59

and dos attack seems to not be working anymore

**Review by aristofanischionis — APPROVED** [review]
Thank you for the great work and completing successfully Google Summer of Code 2024. We are glad we had you as part of our team!
Good luck in your future endeavours.

**Pull request merged**
